### DISCUSSION TITLE
speaker recognition tutorial : all losses diverge :(

### DISCUSSION THREAD

**Discussion by pierrebedu** [NVIDIA-NeMo/NeMo#3219] [General]
Hi,

Following the speaker recognition tutorial, i'm surprised because all losses diverge.

For the first training step it is announced and not annoying.
For the fine tuning step it is not normal.

Any idea to make this converge? thanks

PS : the tutorial is broken due to a torchaudio update. to make this work on google collab, add  this :

!pip install torch==1.10.0+cu111 torchvision==0.11.1+cu111 torchaudio==0.10.0+cu111 -f https://download.pytorch.org/whl/cu111/torch_stable.html

**Comment by okuchaiev** [comment]
@nithinraok can you please look at this

**Comment by nithinraok** [comment]
Thanks for reporting this, let me have a look

**Reply by pierrebedu** [reply]
any clue ? :)

**Reply by nithinraok** [reply]
I just ran and everything seems normal. 
Some suggestions are to make lr high like 0.06 (instead of 0.06) and train for more epochs, this will show you losses converging sooner than later.
Also note: an4 speaker dataset is not designed for speaker recognition, but I have used it for simplicity and ease of use for the tutorial. For indepth tutorial run it on HI-MIA dataset or voxceleb.

**Reply by pierrebedu** [reply]
well thank you very much
i will give it a try :)
